src/middlewared/middlewared/plugins/network_/utils.py

```diff
@@ -0,0 +1,72 @@
+import typing
+
+
+def get_interface_name(item: dict, key: str, interface_dict: dict) -> typing.Optional[str]:
+
+    def get_iface_name(iface_key: str, iface: dict) -> typing.Optional[str]:
+        iface_id = iface[iface_key]
+        if iface_id in interface_dict:
+            return interface_dict[iface_id]['int_interface']
+
+    if key == 'interfaces' and isinstance(item, dict) and 'int_interface' in item:
+        return item['int_interface']
+    elif key == 'bridge':
+        return get_iface_name('interface', item)
+    elif key == 'vlan' and 'vlan_vint' in item:
+        return item['vlan_vint']
+    elif key == 'lagg':
+        return get_iface_name('lagg_interface', item)
+    elif key == 'laggmembers' and 'lagg_interface' in item:
+        return item['lagg_interface']
+    elif key == 'alias' and 'alias_interface' in item:
+        return get_iface_name('alias_interface', item)
+
+
+def find_interface_changes(original_datastores: dict, current_datastores: dict) -> typing.List[str]:
```

Comment:
I'm sorry but no. This function is insanely complex. I don't want to make our network API any more complex than what it already is. This function is unmanageable, frankly. If this is the complexity that is required to send changed events, then we need to not send changed events.